### ISSUE TITLE
[no-func-assign] Duplicates TypeScript's TS2539 error

### ISSUE THREAD

**Issue by OliverSieweke** [typescript-eslint/typescript-eslint#983]
Recommended ESLint rules that are already handled by TypeScript are turned off in `plugin:@typescript-eslint/recommended`. 
The recommended [`no-func-assign`](https://eslint.org/docs/rules/no-func-assign) rule should be included in the list as it triggers an ESLint as well as a TypeScript error.

**Repro**

```JSON
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended"
  ]
}
```

```TS
function foo() {}
foo = bar;
```

**Expected Result**

A simple TypeScript error for line 2:

- `TS2539: Cannot assign to 'foo' because it is not a variable.`

**Actual Result**

A duplicate error for line 2, one coming from TypeScript the other one coming from ESLint: 

- `TS2539: Cannot assign to 'foo' because it is not a variable.`
- `ESLint: 'foo' is a function.(no-func-assign)`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.0` |
| `@typescript-eslint/parser`        | `2.3.0` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `6.4.0` |
| `node`                             | `10.16.3` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
Thanks for the suggestion!
I think it's a good idea to add this to the list.

We are unable to make changes to the recommended sets without a breaking change, so we'll leave this open for now - but I've added it to the list of things to get in with the 3.0 release.

**Comment by bradzacher** [comment]
consolidated eslint-recommended changes in #1273
